plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/StandardDeployPreferencesPanel.java

```diff
@@ -184,8 +185,10 @@ private void setupProjectIdDataBinding(DataBindingContext context) {
     IViewerObservableValue projectList = ViewerProperties.singleSelection().observe(projectSelector.getViewer());
     IObservableValue projectIdModel = PojoProperties.value("projectId").observe(model);
     context.bindValue(projectList, projectIdModel,
-                      new UpdateValueStrategy().setConverter(new GcpProjectToProjectIdConverter()),
-                      new UpdateValueStrategy().setConverter(new ProjectIdToGcpProjectConverter()));
+                      new UpdateValueStrategy().setConverter(new GcpProjectToProjectIdConverter())
+                          .setAfterConvertValidator(new ProjectSelectorValidator(requireValues)),
+                      new UpdateValueStrategy().setConverter(new ProjectIdToGcpProjectConverter())
+                          .setAfterGetValidator(new ProjectSelectorValidator(requireValues)));
```

Comment:
We could use the same instance here, or we could even have two static instance which was created with true/false for `requireValues`, and obtain the appropriate one here through a factory method.